bittensor/chain_data.py

```diff
@@ -225,15 +225,52 @@ class ChainDataType(Enum):
 U16_MAX = 65535
 U64_MAX = 18446744073709551615
 
+@overload
```

Comment:
What's the role of `@overload` here? Is this substantially different from `@abstractmethod`?

Reply:
It allows you to add type hints for multiple signatures of the same function. Meaning for multiple types.
Now that I think about it, this could be a `Union`... instead